lib/simplifyExpression/functionsSearch/nthRoot.js

```diff
@@ -205,54 +206,62 @@ function groupTermsByRoot(node) {
 
   radicandNode.args.sort(sortNodes);
 
-  const terms = radicandNode.args.map(arg => print(arg));
+  const termStrings = radicandNode.args.map(arg => print(arg));
 
-  // Collect groups when there are duplicate terms in the radicand
-  if ([...new Set(terms)].length !== terms.length) {
-    // dictionary storing the number of times a constant appears
-    // e.g. 2 * 2 * 2 => {'2': 3} , 2 appears 3 times
-    const count = {};
 
-    radicandNode.args.forEach(arg => {
-      arg = print(arg);
-      count[arg] ? count[arg]++ : count[arg] = 1;
-    });
-
-    const flatten = arr => arr.reduce(
-      (acc, val) => acc.concat(val), []
-    );
-
-    newNode = Node.Creator.nthRoot(
-      Node.Creator.operator(
-        '*',
-        flatten(Object.keys(count).map(key => {
-          let leftover = count[key];
-          const term = math.parse(key);
-          const times = rootValue;
-
-          const args = [];
-
-          while (leftover - times > 0) {
-            leftover -= times;
-            args.push(Node.Creator.parenthesis(
-              Node.Creator.operator('*', Array(times).fill(term))));
-          }
-          const arg = leftover === 1
-                ? term
-                : Node.Creator.parenthesis(
-                  Node.Creator.operator('*', Array(leftover).fill(term)));
-          args.push(arg);
-
-          return args;
-        }))), rootNode);
-
-    return Node.Status.nodeChanged(
-      ChangeTypes.GROUP_TERMS_BY_ROOT, node, newNode);
-  }
-  else {
-    // if we don't group any factors, then we can't simplify it any more
+  // there is nothing to simplify when there are no duplicates of factors in the radicand
+  if ([...new Set(termStrings)].length === termStrings.length) {
     return Node.Status.noChange(node);
   }
+
+  // dictionary storing the number of times a constant appears
+  // e.g. 2 * 2 * 2 => {'2': 3} , 2 appears 3 times
+  const count = {};
+
+  radicandNode.args.forEach(arg => {
+    const argString = print(arg);
+    count[argString] ? count[argString]++ : count[argString] = 1;
+  });
+
+  const flatten = arr => arr.reduce(
```

Comment:
defining it this far before you use it is kinda confusing

can't you just do this below?

```
// Compress array of arrays
const newTerms = newTermGroups.reduce((acc, val) => acc.concat(val), []);
```